### PR TITLE
fix ‘post_fp_cp2k’, add param rfailed

### PR DESCRIPTION
**Pull request by HuangJiameng**
1. ratio_failed was not applied in post_fp_cp2k.
2. When a cp2k task fails, it returns an empty "output" file. Since dpdispatcher only checks whether the output file exists or not, it marks the task finished and sends back the empty file. An empty or not converged output file leads to errors in the following step. Commenting out check_type_map is a temporary solution, which can run the next step by ignoring empty output files. To solve this problem fundamentally, dpdispatcher should probably be modified as well. 

- add ratio_failed param to cp2k in 02.fp step
- fix a bug in `post_fp_cp2k`

### CODE REVIEW

File: dpgen/generator/run.py [hunk 1]
```diff
@@ -3026,7 +3029,7 @@ def post_fp_cp2k (iter_index,
         all_sys = None
         for oo in sys_output :
             _sys = dpdata.LabeledSystem(oo, fmt = 'cp2k/output')
-            _sys.check_type_map(type_map = jdata['type_map'])
+            #_sys.check_type_map(type_map = jdata['type_map'])
```
Comment (wanghan-iapcm) [inline dpgen/generator/run.py:3032]: Could you explain why the check_type_map is commented out?

Comment (HuangJiameng) [reply dpgen/generator/run.py:3032]: Thanks for your comment, I have edited my comment.

**Review by wanghan-iapcm — APPROVED** [review]

**Comment by wanghan-iapcm** [comment]
> 2\. When a cp2k task fails, it returns an empty "output" file. Since dpdispatcher only checks whether the output file exists or not, it marks the task finished and sends back the empty file. An empty or not converged output file leads to errors in the following step. Commenting out check_type_map is a temporary solution, which can run the next step by ignoring empty output files. To solve this problem fundamentally, dpdispatcher should probably be modified as well.

@njzjz @felix5572 FYI

**Comment by njzjz** [comment]
> > 2. When a cp2k task fails, it returns an empty "output" file. Since dpdispatcher only checks whether the output file exists or not, it marks the task finished and sends back the empty file. An empty or not converged output file leads to errors in the following step. Commenting out check_type_map is a temporary solution, which can run the next step by ignoring empty output files. To solve this problem fundamentally, dpdispatcher should probably be modified as well.
> 
> @njzjz @felix5572 FYI

This is not the expected behavior. dpdispatcher should check the return code, unless cp2k always returns 0.

**Pull request merged**